src/map/utils/battleutils.cpp

```diff
@@ -2222,6 +2222,8 @@ namespace battleutils
 
                 case TYPE_PET:
                     ((CPetEntity*)PDefender)->loc.zone->UpdateEntityPacket(PDefender, ENTITY_UPDATE, UPDATE_COMBAT);
+                    // pets should lost enmity as well from normal attacks
+                    ((CMobEntity*)PAttacker)->PEnmityContainer->UpdateEnmityFromAttack(PDefender, damage);
```

Comment:
I think we may need to add the same call after line 2218 to account for BST charmed pets

```
if (PDefender->PMaster != nullptr && PDefender->PMaster->objtype == TYPE_PC)
{
	((CPetEntity*)PDefender)
		->loc.zone->UpdateEntityPacket(PDefender, ENTITY_UPDATE, UPDATE_COMBAT);
	((CMobEntity*)PAttacker)->PEnmityContainer->UpdateEnmityFromAttack(PDefender, damage);
}
```
Otherwise yessss, this is exactly the kind of change I was considering as well. Nice work!

Reply:
> I think we may need to add the same call after line 2218 to account for BST charmed pets
> 
> ```
> if (PDefender->PMaster != nullptr && PDefender->PMaster->objtype == TYPE_PC)
> {
> 	((CPetEntity*)PDefender)
> 		->loc.zone->UpdateEntityPacket(PDefender, ENTITY_UPDATE, UPDATE_COMBAT);
> 	((CMobEntity*)PAttacker)->PEnmityContainer->UpdateEnmityFromAttack(PDefender, damage);
> }
> ```
> 
> Otherwise yessss, this is exactly the kind of change I was considering as well. Nice work!

Yep, I added also there and included some checking for type to make sure. :)